src/uint/modular/runtime_mod/runtime_pow.rs

```diff
@@ -35,8 +39,72 @@ impl<const LIMBS: usize> DynResidue<LIMBS> {
     }
 }
 
-impl<const LIMBS: usize> PowBoundedExp<Uint<LIMBS>> for DynResidue<LIMBS> {
-    fn pow_bounded_exp(&self, exponent: &Uint<LIMBS>, exponent_bits: usize) -> Self {
+impl<const LIMBS: usize, const RHS_LIMBS: usize> PowBoundedExp<Uint<RHS_LIMBS>>
+    for DynResidue<LIMBS>
+{
+    fn pow_bounded_exp(&self, exponent: &Uint<RHS_LIMBS>, exponent_bits: usize) -> Self {
         self.pow_bounded_exp(exponent, exponent_bits)
     }
 }
+
+impl<const N: usize, const LIMBS: usize, const RHS_LIMBS: usize>
+    MultiExponentiateBoundedExp<Uint<RHS_LIMBS>, [(Self, Uint<RHS_LIMBS>); N]>
+    for DynResidue<LIMBS>
+{
+    fn multi_exponentiate_bounded_exp(
+        bases_and_exponents: &[(Self, Uint<RHS_LIMBS>); N],
+        exponent_bits: usize,
+    ) -> Self {
+        // TODO: is it fine? will crash if N == 0
```

Comment:
Something like this, but unfortunately this does run into the "can't use generic parameters from outer item" error:

```suggestion
        const _: () = assert!(N != 0, "N cannot be zero");
```

I saw some blog post that showed a workaround recently, but now I can't find it anymore.

Reply:
Absent a solution like this, you can panic for now with a TODO to replace the runtime panic with a const assert/bounds, e.g. using `generic_const_exprs`

Reply:
Oh, apparently there is a hacky way to make a const assertion work: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b522f72bebe0140e1302894391107903

Reply:
Opened a PR to add some macros that implement such assertions in a much clearer way: https://github.com/RustCrypto/crypto-bigint/pull/293

Reply:
Okay, with #293 merged, if you rebase, this should now work:

```suggestion
        const_assert_ne!(N, 0, "bases_and_exponents must not be empty");
```